package/MDAnalysis/core/groups.py

```diff
@@ -822,11 +822,13 @@ def rotate(self, R, point=(0, 0, 0)):
         point = np.asarray(point)
 
         # changes the coordinates (in place)
-        self.translate(-point)
-        x = self.atoms.unique.universe.trajectory.ts.positions
-        idx = self.atoms.unique.indices
+        if not np.allclose(point, np.zeros(3)):
```

Comment:
It feels wrong to create `np.zeros(3)` twice but I know that in the grand scheme of things it will be negligible speedup to do something like
```python
is_at_origin = np.allclose(point, np.zeros(3))
if not is_at_origin:
   self.translate(-point)
...
if not is_at_origin:
   self.translate(point)
```